WordPress/Classes/ViewRelated/Post/PostListViewController.m

```diff
@@ -0,0 +1,1319 @@
+        navController.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
+        navController.modalPresentationStyle = UIModalPresentationFormSheet;
+    }
+
+    [self presentViewController:navController animated:YES completion:nil];
+}
+
+
+#pragma mark - TableView Handler Delegate Methods
+
+- (NSManagedObjectContext *)managedObjectContext
+{
+    return [[ContextManager sharedInstance] mainContext];
+}
+
+- (NSFetchRequest *)fetchRequest
+{
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:NSStringFromClass([Post class])];
+    fetchRequest.predicate = [self predicateForFetchRequest];
+    NSSortDescriptor *sortDescriptorDate = [NSSortDescriptor sortDescriptorWithKey:@"date_created_gmt" ascending:NO];
+    fetchRequest.sortDescriptors = @[sortDescriptorDate];
+    fetchRequest.fetchBatchSize = PostsFetchRequestBatchSize;
+    return fetchRequest;
+}
+
+- (void)updateAndPerformFetchRequestClearingCachedRowHeights:(BOOL)clearCachedRowHeights
+{
+    NSAssert([NSThread isMainThread], @"PostsViewController Error: NSFetchedResultsController accessed in BG");
+
+    NSPredicate *predicate = [self predicateForFetchRequest];
+    NSError *error = nil;
+    [self.tableViewHandler.resultsController.fetchRequest setPredicate:predicate];
+    [self.tableViewHandler.resultsController performFetch:&error];
+    if (error) {
+        DDLogError(@"Error fetching posts after updating the fetch request predicate: %@", error);
+    }
+    if (clearCachedRowHeights) {
+        [self.tableViewHandler clearCachedRowHeights];
+    }
+    [self.tableView reloadData];
+    [self configureNoResultsView];
+}
+
+- (NSPredicate *)predicateForFetchRequest
+{
+    NSMutableArray *predicates = [NSMutableArray array];
+
+    NSPredicate *basePredicate = [NSPredicate predicateWithFormat:@"blog = %@ && original = nil", self.blog];
+    [predicates addObject:basePredicate];
+
+    NSPredicate *filterPredicate = [self currentPostListFilter].predicateForFetchRequest;
+    [predicates addObject:filterPredicate];
+
+    if (self.blog.isMultiAuthor && ![self shouldShowPostsForEveryone] && [self.blog.account.userID integerValue] > 0) {
+        NSPredicate *authorPredicate = [NSPredicate predicateWithFormat:@"authorID = %@", self.blog.account.userID];
+        [predicates addObject:authorPredicate];
+    }
+
+    NSString *searchText = self.searchController.searchBar.text;
+    if ([searchText length] > 0) {
+        NSPredicate *searchPredicate = [NSPredicate predicateWithFormat:@"postTitle CONTAINS %@", searchText];
+        [predicates addObject:searchPredicate];
+    }
+
+    NSPredicate *predicate = [NSCompoundPredicate andPredicateWithSubpredicates:predicates];
+    if ([self.recentlyTrashedPostIDs count] > 0) {
+        NSPredicate *trashedPredicate = [NSPredicate predicateWithFormat:@"postID IN %@", self.recentlyTrashedPostIDs];
+        predicate = [NSCompoundPredicate orPredicateWithSubpredicates:@[predicate, trashedPredicate]];
```

Comment:
Not sure if this is intentional, but if you trash a post and then search for something else, the trashed post still shows up

![screen shot 2015-04-28 at 13 41 07](https://cloud.githubusercontent.com/assets/8739/7368818/56c48c14-edac-11e4-9e02-2360b6efd7a9.png)